promql/engine.go

```diff
@@ -1455,6 +1457,66 @@ func (ev *evaluator) rangeEvalAgg(ctx context.Context, aggExpr *parser.Aggregate
 	return result, warnings
 }
 
+// expandSeriesToMatrix expands vs.Series into a Matrix.
```

Comment:
Please mention what effect `start`, `end` and `interval` have on the operation.
You may find that "expand" is not fully descriptive of what is going on.

Reply:
Thanks, will do.

Reply:
I've documented the method more extensively, and ended up renaming it to `generateMatrix`, since I agree it's not a straight expansion of the series iterators.